src/main/java/com/powsybl/network/hypothesis/GeneratorsStartupAlgorithm.java

```diff
@@ -0,0 +1,274 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.network.hypothesis;
+
+import com.powsybl.iidm.network.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ * @author Chamseddine BENHAMED <chamseddine.benhamed at rte-france.com>
+ */
+public class GeneratorsStartupAlgorithm {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GeneratorsStartupAlgorithm.class);
+
+    List<Generator> startupGroupsPowerMax = new ArrayList<>();
+
+    double nuclearBandSetting = 0.05;
+    double thermalBandSetting = 0.05;
+    double hydroBandSetting = 0.1;
+
+    double lossCoefficient = 0;
+    double defaultAbatementCoefficient = 0;
+
+    StartupMarginalGroupType startUpMarginalGroupType = StartupMarginalGroupType.CLASSIC;
+
+    //thresholds for voltage adjustment
+    double pThreshold = 0.0;
+    double qThreshold = 0.0;
+
+    /**
+     * starts generators groups using Classic or Mexico algorithm
```

Comment:
It would be great to define in what context that can be useful and the unfolding of the algorithm in plain English